tapa/verilog/xilinx/module.py

```diff
@@ -259,8 +265,10 @@ def get_port_of(self, fifo: str, suffix: str) -> IOPort:
                 if port is not None:
                     _logger.warning("assuming %s is a singleton array", singleton_fifo)
                     return port
-        msg = f"module {self.name} does not have port {fifo}.{suffix}"
-        raise ValueError(msg)
+        all_ports = list(self.ports.keys())
+
+        msg = f"module {self.name} does not have port {fifo}.{suffix} {all_ports!s}"
+        raise self.NoMatchingPortError(msg)
```

Comment:
Change to `Module.NoMatchingPortError` to explicitly show that it is from the class.